python/lsst/ts/hvac/hvac_utils.py

```diff
@@ -0,0 +1,33 @@
+# This file is part of ts_hvac.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+__all__ = ["to_camel_case"]
+
+
+def to_camel_case(string, first_lower=False):
+    cc = ""
+    parts = string.split("_")
+    for part in parts:
+        cc += part[0].upper() + part[1:].lower()
```

Comment:
This fails if the string is blank and I fear the error will be confusing. Please either fix it to work or test for the condition and raise a clearer error.

This is also rather inefficient for two reasons:
* Accumulating a string is less efficient than using the `join` str method.
* The capitalize str method does the main part of what you want; no need to split each `part`.

I suggest something like the following (which silently returns "" for a blank str; add a test if you want to complain about that case):
```
def to_camel_case(string, first_lower=False):
    """Return a CamelCase or camelCase version of a snake_case str.""" 
    if first_lower:
        first, _, rest = string.partition("_")
    else:
        first, rest = ("", string)
    return first.lower() + "".join(part.capitalize() for part in rest.split("_"))
```
In any case I suggest partition instead of split because it returns an empty string if the separator is not found, which handles the empty string case neatly. But if you explicitly reject an empty string then you can partition using `split("_", 1)` if you prefer.

Reply:
Thanks, this indeed is much cleaner.